webhook/resourcesemantics/validation/validation_admit.go

```diff
@@ -36,25 +36,51 @@ import (
 var errMissingNewObject = errors.New("the new object may not be nil")
 
 // Callback is a generic function to be called by a consumer of validation
+// To preserve backwards compatibility, it also doubles up as a way to configure
+// which verbs and which subresources are configured in the
+// validatingwebhookconfiguration. For configuring which verbs and subresources
+// should get validated, create the Callback and pass it to
```

Comment:
I don't think we should utilize the struct in this way

Reply:
I think we need an additional `New` method that's more generation and allows for more options